File: Objetivo 7/ExcecaoJava/src/control/AppFileMyExceptionsClass.java
package control;

/**
 *
 * @author vagner
 */
public class AppFileMyExceptionsClass {
    public static void main(String[] args) {
        try{
            //abrir(""); //sem argumento
            
            abrir("/tmp/teste.txt"); //impossível ler o arquivo
        
        } catch (ParameterException | FileNotFoundException ex) {
            System.out.println("\nEntrou no catch.");
            System.out.println(ex.getCause() + " ### " + ex.getMessage() + " ### " + ex.getClass());
        }finally{
            System.out.println("\nSempre executa o que estiver no finally.");
        }
    }
    
    private static void abrir(String caminho) throws ParameterException, FileNotFoundException{
        
        if(caminho.equals("")){
            throw new ParameterException();
        }
        if(!caminho.equals("")){
            throw new FileNotFoundException();
        }
    
    }//fim método
}

class ParameterException extends Exception{}
class FileNotFoundException extends Exception{} 

File: Objetivo5/ExcecaoPHP/control/AppFileMyExceptionsClass.php
<?php

function abrir($file = null){
    if(!$file){
        throw new ParameterException("sem argumento", 1, null);
    }
    if(!file_exists($file)){
        throw new FileNotFoundException("arquivo inexistente", 2, null);
    }
    if(!$retorno = file_get_contents($file)){
        throw new FilePermissionException("impossível ler o arquivo", 3, null);
    }
    
    return $retorno;
    
}//fim function

//definição da subclasses de Exception
class ParameterException extends Exception{}
class FileNotFoundException extends Exception{}
class FilePermissionException extends Exception{}

try{
    //$arquivo = abrir(); //sem argumento (ParameterException)
    $arquivo = abrir("/home/vagner/teste.txt"); //arquivo inexistente (FileNotFoundException)
    //$arquivo = abrir("/tmp/texte.txt"); //impossível ler o arquivo (FilePermissionException)
} catch (ParameterException $ex) { //Mudar o tipo para ver o efeito
    echo $ex->getFile() . ' ### ' . $ex->getLine() . ' ### ' . $ex->getMessage() . ' ### ' . $ex->getCode(); //. ' ### ' . print_r($ex->getTrace());
} catch (FileNotFoundException $ex) { //Mudar o tipo para ver o efeito
    echo $ex->getFile() . ' ### ' . $ex->getLine() . ' ### ' . $ex->getMessage() . ' ### ' . $ex->getCode(); // . ' ### ' . print_r($ex->getTrace());
} catch (FilePermissionException $ex) { //Mudar o tipo para ver o efeito
    echo $ex->getFile() . ' ### ' . $ex->getLine() . ' ### ' . $ex->getMessage() . ' ### ' . $ex->getCode(); // . ' ### ' . print_r($ex->getTrace());
}finally{
    echo "\nSempre executa o que estiver no finally.";
}

